source/fab/database.py

```diff
@@ -22,8 +22,11 @@ def __init__(self, filename: Path, adler32: int):
         self.adler32 = adler32
 
     def __eq__(self, other):
+        if not isinstance(other, FileInfo):
+            message = f"Cannot compare FileInfo with {other.__class__}"
+            raise ValueError(message)
         return (str(other.filename) == str(self.filename)) \
```

Comment:
I realise this isn't strictly related to this change, but while I was checking this part it occurred to me... if `other.filename` and `self.filename` are identical, aren't we _literally_ talking about the exact same file on disk?  In which case would we ever expect their `adler32` hashes to be different?

Reply:
That is true if they are referring to the same file at the same instant in time. However if they refer to the same file at different points in time they could have different contents.

Reply:
Ah okay I think I see what you're saying - it's dependent upon when exactly the file instances were created (and their hashes computed)